src/yargs-params.js

```diff
@@ -9,39 +9,86 @@
  * OF ANY KIND, either express or implied. See the License for the specific language
  * governing permissions and limitations under the License.
  */
+const path = require('path');
+const fse = require('fs-extra');
+const dotenv = require('dotenv');
+
+/**
+ * Decoded the params string. First as JSON and if this fails, as ENV format.
+ *
+ * @param {string} params Params string.
+ * @param {string} cwd CWD of to resolve file, assumes that `param` is a file path.
+ * @returns {*} Decoded params object.
+ */
+function decodeParamString(params, cwd = '') {
+  let content = params.trim();
+  if (cwd) {
+    const filename = path.resolve(cwd, content);
+    if (!fse.existsSync(filename)) {
+      throw Error(`Specified param file does not exist: ${path.relative(process.cwd(), filename)}`);
+    }
+    content = fse.readFileSync(filename, 'utf-8').trim();
+  }
+  let data;
+  if (typeof params === 'object') {
```

Comment:
How can `params` be an object here ? If it is an object, the previous `params.trim()` would most likely fail, no ?
Unit test those functions would reveal the flaws...

Reply:
good catch....copy/paste leftover from another project ;-)